framework/Optimizers/GradientBasedOptimizer.py

```diff
@@ -958,9 +958,11 @@ def writeToSolutionExport(self,traj, recent, accepted, overwrite=None):
   def checkResampleOption(self,traj):
     """
       Turn on self.resample[traj] while checking self.resampleSwitch.
-      This method is equivalent to self.resample[traj] = resampleSwitch while needed
+      This method is equivalent to self.resample[traj] = self.resampleSwitch while needed
       @ In, traj, int, the trajectory for which an entry is being written
-      @ Out, bool, True if resample switch is on
+      @ Out, bool, return self.resampleSwitch, True if resample switch is on
```

Comment:
`@ Out, <varname or methodname>, <type>, <description>`

I think you're missing the (admittedly confusing) output variable name